dascore/proc/basic.py

```diff
@@ -606,3 +610,36 @@ def pad(
     new_coords = patch.coords.update(**new_coords)
 
     return patch.new(data=new_data, coords=new_coords)
+
+
+@patch_function()
+def roll(patch, samples=False, **kwargs):
+    """
+    Roll patch array elements along a given dimension.
+
+    Parameters
+    ----------
+    patch
+        input patch
+    samples
+        If True value indicates coordinate or value of dimension
+    **kwargs
+        specifies dimension and number of elements to roll
+
+    Examples
+    --------
+    >>> import dascore as dc
+    >>> patch = dc.get_example_patch()
+    >>> # roll time dimension 5 elements
+    >>> rolled_patch = patch.roll(time=5, samples=True)
+    >>> # roll distance dimension 30 meters(or units of distance in patch)
+    >>> rolled_patch2 = patch.roll(distance=30, samples=False)
+    """
+    dim, axis, input_value = get_dim_value_from_kwargs(patch, kwargs)
+    arr = patch.data
+    coord = patch.get_coord(dim)
+    value = coord.get_sample_count(input_value, samples=samples)
+
+    roll_arr = np.roll(arr, value, axis=axis)
+
+    return patch.new(data=roll_arr)
```

Comment:
Do we need to update the coord as well? Or maybe just adding an optional update_coord argument so when is set True, the function will roll the coord as well?

Reply:
> Do we need to update the coord as well? Or maybe just adding an optional update_coord argument so when is set True, the function will roll the coord as well

Ya, good point. Normally `Patch.roll` would be used to create a new patch that will broadcast with the old one. For example, in cross correlation. In this case you wouldn't want to coords to update, you are purposefully "rolling" the data around stationary coordinates. However, I could also see a case where you might want to roll the coords as well, so an optional argument to do so makes sense.

Reply:
@Nik-P2, do you think you can implement this? 

You need to add an optional keyword called `update_coord` or `roll_coord` which is `False` by default. Then if it is `True` do something like this (haven't tested but this should be enough to get you started):

```python
...
if roll_coord:  # Handle rolling coordinate if needed
    coord = patch.get_coord(dim)
    rolled_coord_array = np.roll(coord.values, value, axis)
    new_coord = coord.update(values=rolled_coord_array)
    patch = patch.update_coords(**{dim: new_coord})

return patch.new(data=roll_arr)
```

Make sure to write another test case that runs this logic.

Reply:
Yup, I will work on that now.

Reply:
Looks great. Thanks, @Nik-P2 !